src/matchers/mock/toBeRequestedWith.ts

```diff
@@ -0,0 +1,288 @@
+import { waitUntil, enhanceError } from '../../utils'
+import { runExpect } from '../../util/expectAdapter'
+import { equals } from '../../jasmineUtils'
+
+const STR_LIMIT = 80
+
+function toBeRequestedWithFn(
+    received: WebdriverIO.Mock,
+    requestedWith: ExpectWebdriverIO.RequestedWith = {},
+    options: ExpectWebdriverIO.CommandOptions = {}
+): any {
+    const isNot = this.isNot || false
+    const { expectation = 'called with', verb = 'be' } = this
+
+    return browser.call(async () => {
+        let actual: WebdriverIO.Matches | undefined
+        const pass = await waitUntil(
+            async () => {
+                for (const call of received.calls) {
+                    actual = call
+                    if (
+                        methodMatcher(call.method, requestedWith.method) &&
+                        urlMatcher(call.url, requestedWith.url) &&
+                        headersMatcher(call.headers, requestedWith.headers) &&
+                        bodyMatcher(call.postData, requestedWith.request) &&
+                        bodyMatcher(call.body, requestedWith.response)
+                    ) {
+                        return true
+                    }
+                }
+
+                return false
+            },
+            isNot,
+            { ...options, wait: isNot ? 0 : options.wait }
+        )
+
+        const message = enhanceError(
+            'mock',
+            minifyRequestedWith(requestedWith),
+            minifyRequestMock(actual, requestedWith) || 'was not called',
+            this,
+            verb,
+            expectation,
+            '',
+            options
+        )
+        return {
+            pass,
+            message: (): string => message,
+        }
+    })
+}
+
+/**
+ * is actual method matching an expected method or methods
+ */
+const methodMatcher = (method: string, expected?: string | Array<string>) => {
+    if (typeof expected === 'undefined') {
+        return true
+    }
+    if (!Array.isArray(expected)) {
+        expected = [expected]
+    }
+    return expected
+        .map((m) => {
+            if (typeof m !== 'string') {
+                return console.error('expect.toBeRequestedWith: unsupported value passed to method ' + m)
```

Comment:
Do we not want to throw or want this matcher to fail if this happens?

Reply:
Initially, I implemented it to throw an exception, however [`waitUntil` suppresses](https://github.com/webdriverio/expect-webdriverio/blob/master/src/utils.ts#L27) errors like this.

So I decided that it's still better to print an error rather than hiding it.

Reply:
Gotcha, will this make the assertion fail? Should we do the type check at an earlier stage to be able to throw?

Reply:
1. yes, the assertion will fail
2. I believe so. Type check might be helpful not only for `method`. Let me raise an issue to implement a better type check for the matcher

Reply:
https://github.com/webdriverio/expect-webdriverio/issues/199